packages/augur-core/source/contracts/Augur.sol

```diff
@@ -113,7 +112,7 @@ contract Augur is IAugur, IAugurCreationDataGetter, CashSender {
         require(registry[_key] == address(0), "Augur.registerContract: key has already been used in registry");
         require(_address.exists());
         registry[_key] = _address;
-        if (_key == "ShareToken" || _key == "MarketFactory") {
+        if (_key == "ShareToken" || _key == "MarketFactory" || _key == "EthExchange") {
```

Comment:
I understand why this line is here, but this is also the first place where Orders code/trust (ZeroX orders in this case) bleeds into the rest of Augur. Before this, it was really kind of bolted onto the side

Reply:
Yea its unfortunate. We _could_ avoid this by having yet another approval for the EthExchange though I'd rather not if possible. The reason I think this is alright is that while only the trading code _looks_ at the ETH exchange its not strictly a part of trading. Really it's its own thing.